chore: TS refactor to composite project, upgrade to TS 4.5

Nodonisko opened the pull request:
**Changelog:**

**Global:**
- Tuned `package.json` scripts - nicely divide into sections, removed unused, added some aliases
- Upgraded to TS 4.5
- Migrated to TS composite project (https://www.typescriptlang.org/docs/handbook/project-references.html). Some benefits of this:
- - Each file is now compiled just once witch proper config, previously was always used `tsconfig.json` from project where file was imported and file was compiled multiple times if was used in multiple project.
- - That brings us small performance benefit, that will be more significant when repo will grow.
- - Caching is now enabled for TS so `type-check` re-runs are really fast (just few sec) 🚀 
- - All packages now uses this composite mode for `type-check`, packages with `tsconfig.lib.json` still build like previously.
- - All packages now use `--build` instead of `--project`, it will work same except that if there are `references`, referenced project will be builded.
- Changes prettier config to `printWidth: 50`, it's just nicer and works better with references generator.
- Added new package `scripts` to repo root, this package will be home for some mainteneance scripts. In future we can move more scripts here, also some from `suite-data/scripts` since some of them causing module circular dependency (check comment in `packages/suite-data/scripts/list-duplicates.ts`) cc @mroz22. 
- Added `run-script` script to root `package.json` which will be used for running all sort of scripts, with support for scripts with ESM imports.
- Added `update-project-references` script to root `package.json` which will automatically update `tsconfig.json` project references for packages, according to deps in their `package.json`.
- Added `type-check:force` that will drop TSC cache before `type-check`.
- Created default `tsconfig.json` with recommended options in repo root that should all packages extend.
- Moved aliases to special `tsconfig.aliases.json` that should be used only in packages that need aliases to work. **It should not be used in any new package.**
- Created new default `tsconfig.lib.json` in root that is used to extend `tsconfig.lib.json` in packages.

**blockchain-link:**
- Better types in `tsconfig.json` by usage `library: ["webworker"]` with combination `types: ["web"]`. Thanks to that `tsconfig.webworkers.json` could be completely removed. And also `type-check` and intelisense is much precise for webworker types.
- Ignored `lib/ui` in `files` in `package.json` because `tsconfig.lib.json` `includes` was changed to include folder `src`, because everything could be build by same `tsconfig.json`

**components**
- Hot module replacement works 🎉 
- Removed unnecessary build step
- Lot of import/export type fixes there that was needed for `isolatedModules` TS options

**connect-iframe**
- Added babel plugin to be able to compile TS from `utils` package.

`suite-data`
- Added `type-check` for this package. That revealed one problem with circle deps here (check comment in `packages/suite-data/scripts/list-duplicates.ts`).
- Removed some imports from `global.d.ts` which was not needed anymore

**suite-desktop**
- This package have to wait until #4852 wil be merged and then some changes will be needed or maybe some changes won't be neccesary after that. So I recommend skip this package during code review until it will be rebased.
- Force `build.js` to bundle `@trezor/utils` which need to be build because of TS and also could be tree shaken during build.
- Removed some unused imports from `global.d.ts`

**suite-native**
- Disabled `type-check` for now. Lot more changes will be needed here to make TS works since aliases (incl. babel aliases) are no option anymore. I already started work on refactoring this.

**suite-storage**
- Removed unccessary build
- Pointed `main` to `web`, this package will be refactored to use `.native.ts` extensions.
- Added to `@ts-ignore` to one place in `src/web/index.ts` since TS upgrade caused some type change for DB keys. This should be investigated more.

**suite-web-landing**
- Remove unused babel aliases, Next can read them from `tsconfig`.
- Removed some usage of suite aliases, since there was few of them.
- Remove some unused type reference in `global.d.ts`
- Add `components`  Next transpile modules because they are now in TS.
- Upgrade to Next 12

**suite-web**
- Removed unused import from `global.d.ts`
- Removed few aliases

**suite**
- Migrate Jest to `babel-jest`. `ts-jest` is rudundant because because tests are type-checked by TSC anyway. Also `ts-jest` still has no support for composite projects.
- On few places `setTimeout` => `window.setTimeout`, because there is some problem with Node types that has different return type than number for `setTimeout`
- On few places removed unused export from `export initalState` that was causing TSC to suffer from some infinite recursive type, that will exceed max type lenght for exported types
- Skip `utils/suite/__tests__/env.test.ts` because there is some problem with mocking, check file for details.

**utils**
- Set `sideEffect: false` to `package.json` which will make tree shaking works
- Build output for `tsconfig.lib.json` set to use `module: 'commonjs'` because it will work everywhere, but for example tree shaking won't work. Import directly from `@trezor/suite/lib/...` if you want commonjs

**Possible follow-ups:**
- Why is TS 3.9 still in node_modules?
- Figure out how to publish `@trezor/utils` and don't break HMR and tree shaking (already have few ideas, some them should work).
- Add generator for creating new packages with some recommended setup to avoid copy&paste from random packages.

matejkriz reviewed (changes requested):
integration tests for
- transport https://gitlab.com/satoshilabs/trezor/trezor-suite/-/jobs/2082580394
- rollout https://gitlab.com/satoshilabs/trezor/trezor-suite/-/jobs/2079951842
- and suite desktop https://gitlab.com/satoshilabs/trezor/trezor-suite/-/jobs/2079951841

are failing

Nodonisko commented:
@matejkriz Hopefully fixed, `commonjs` modules need imports from `@trezor/utils/lib`. Can you run integration tests again?

szymonlesisz commented on packages/suite-web/src/support/Router.tsx:
```diff
@@ -1,8 +1,8 @@
-import routes from '@suite-constants/routes';
-import { BundleLoader } from '@suite-components';
+import routes from '@trezor/suite/src/constants/suite/routes';
+import { BundleLoader } from '@trezor/suite/src/components/suite';
```
code history hygiene: this and following import changes belongs to previous commit (removing aliases)

Nodonisko replied:
I removed this files completely from PR, this change is not needed for this PR and should be done in other PR or when we will have replacement do aliases.

szymonlesisz commented on packages/suite-web/src/support/CypressExportStore.tsx:
```diff
@@ -1,5 +1,5 @@
 import { useEffect } from 'react';
-import { Store } from '@suite-types';
+import { Store } from '@trezor/suite/src/types/suite';
```
this belongs to previous commit

Nodonisko replied:
I removed this files completely from PR, this change is not needed for this PR and should be done in other PR or when we will have replacement do aliases.

szymonlesisz commented on packages/suite-web/src/Main.tsx:
```diff
@@ -3,24 +3,24 @@ import { Provider as ReduxProvider } from 'react-redux';
+import ErrorBoundary from '@trezor/suite/src/support/suite/ErrorBoundary';
+import RouterHandler from '@trezor/suite/src/support/suite/Router';
+import ThemeProvider from '@trezor/suite/src/support/suite/ThemeProvider';
+import history from '@trezor/suite/src/support/history';
```
this belongs to previous commit

Nodonisko replied:
I removed this files completely from PR, this change is not needed for this PR and should be done in other PR or when we will have replacement do aliases.

szymonlesisz commented on packages/suite-web-landing/utils/switchFavicon.ts:
```diff
@@ -1,4 +1,4 @@
-import { resolveStaticPath } from '@suite-utils/build';
+import { resolveStaticPath } from '@trezor/suite/src/utils/suite/build';
```
this belongs to previous commit

Nodonisko replied:
I removed this files completely from PR, this change is not needed for this PR and should be done in other PR or when we will have replacement do aliases.

szymonlesisz commented on packages/suite-web-landing/pages/start.tsx:
```diff
@@ -4,9 +4,9 @@ import Layout from '../components/Layout';
-import { URLS } from '@suite-constants';
+import { resolveStaticPath } from '@trezor/suite/src/utils/suite/build';
+import Metadata from '@trezor/suite/src/components/suite/Metadata';
+import { URLS } from '@trezor/suite/src/constants/suite';
```
this belongs to previous commit

Nodonisko replied:
I removed this files completely from PR, this change is not needed for this PR and should be done in other PR or when we will have replacement do aliases.

szymonlesisz commented on packages/suite-web-landing/pages/index.tsx:
```diff
@@ -4,9 +4,9 @@ import Layout from '../components/Layout';
-import { URLS } from '@suite-constants';
+import { resolveStaticPath } from '@trezor/suite/src/utils/suite/build';
+import Metadata from '@trezor/suite/src/components/suite/Metadata';
+import { URLS } from '@trezor/suite/src/constants/suite';
```
this belongs to previous commit

szymonlesisz commented on packages/suite-web-landing/pages/_document.tsx:
```diff
@@ -1,10 +1,10 @@
-import { isEnabled } from '@suite-utils/features';
-import { TOR_URLS } from '@suite-constants/tor';
+import { isEnabled } from '@trezor/suite/src/utils/suite/features';
+import { TOR_URLS } from '@trezor/suite/src/constants/suite/tor';
```
this belongs to previous commit

szymonlesisz commented on packages/suite-web-landing/components/Translation/index.tsx:
```diff
@@ -1,8 +1,8 @@
 import HelperTooltip, {
     Props as HelperTooltipProps,
-} from '@suite-components/Translation/components/HelperTooltip';
+} from '@trezor/suite/src/components/suite/Translation/components/HelperTooltip';
```
this belongs to previous commit

szymonlesisz commented on packages/suite-web-landing/components/FromMytrezorBanner/index.tsx:
```diff
@@ -1,7 +1,7 @@
 import { Icon, colors, variables, Button, H1, H2 } from '@trezor/components';
 import styled from 'styled-components';
-import CollapsibleBox from '@suite-components/CollapsibleBox';
+import CollapsibleBox from '@trezor/suite/src/components/suite/CollapsibleBox';
```
this belongs to previous commit

szymonlesisz commented on packages/suite-web-landing/components/Download/index.tsx:
```diff
@@ -4,8 +4,8 @@ import { Button, Dropdown, Icon, colors, variables, Link } from '@trezor/compone
-import { normalizeVersion } from '@suite-utils/build';
-import { SL_SIGNING_KEY, WIKI_HOW_TO_RUN } from '@suite-constants/urls';
+import { normalizeVersion } from '@trezor/suite/src/utils/suite/build';
+import { SL_SIGNING_KEY, WIKI_HOW_TO_RUN } from '@trezor/suite/src/constants/suite/urls';
```
this belongs to previous commit

szymonlesisz commented on packages/integration-tests/projects/suite-web/support/utils/test-env.ts:
```diff
@@ -1,4 +1,4 @@
-import { Action } from '@suite-types';
+import { Action } from '@trezor/suite/src/types/suite';
```
this belongs to previous commit

szymonlesisz commented on packages/integration-tests/projects/suite-web/support/commands.ts:
```diff
@@ -3,7 +3,7 @@
 import TrezorConnect, { Device, Features } from 'trezor-connect';
 import 'cypress-wait-until';
-import { Store, Action } from '@suite-types';
+import { Store, Action } from '@trezor/suite/src/types/suite';
```
this belongs to previous commit

szymonlesisz commented on packages/suite/src/reducers/suite/resizeReducer.ts:
```diff
@@ -1,5 +1,5 @@
 import produce from 'immer';
-import * as variables from '@trezor/components/lib/config/variables'; // can't import from index cause it would import all UI components
+import * as variables from '@trezor/components/src/config/variables'; // can't import from index cause it would import all UI components
```
this belongs to previous commit

szymonlesisz commented on packages/suite/src/reducers/wallet/graphReducer.ts:
```diff
@@ -6,7 +6,7 @@ import { GraphRange, GraphScale, AccountIdentifier, GraphData } from '@wallet-ty
 import { SETTINGS } from '@suite-config';
 
-interface State {
+export interface State {
```
this export (and following) was never used before. is it required to export them?

Nodonisko replied:
Every export I added anywhere is because TS complained, it's probably because I added `declaration` and `declarationMap` everywhere. Second option is needed because it will make work some additional editor features like Go To Definition etc.

szymonlesisz commented on packages/suite/src/reducers/wallet/coinmarketReducer.ts:
```diff
@@ -70,7 +70,7 @@ interface Sell {
 }
 
-interface State {
+export interface State {
```
export not used?

szymonlesisz commented on packages/suite/src/reducers/suite/suiteReducer.ts:
```diff
@@ -19,7 +19,7 @@ export interface AutodetectSettings {
 }
 
-interface Flags {
+export interface Flags {
```
export not used?

szymonlesisz commented on packages/suite/src/utils/suite/theme.ts:
```diff
@@ -1,4 +1,4 @@
-import { THEME } from '@trezor/components/lib/config/colors';
+import { THEME } from '@trezor/components/src/config/colors';
```
this belongs to previous commit

szymonlesisz commented on packages/suite/src/reducers/wallet/sendFormReducer.ts:
```diff
@@ -4,7 +4,7 @@ import { ACCOUNT, SEND } from '@wallet-actions/constants';
 import { FormState, PrecomposedTransactionFinal } from '@wallet-types/sendForm';
 
-interface SendState {
+export interface SendState {
```
export not used?

szymonlesisz commented on packages/suite/src/components/suite/modals/QrScanner/index.tsx:
```diff
@@ -98,7 +98,7 @@ type Props = Omit<Extract<UserContextPayload, { type: 'qr-reader' }>, 'type'> &
 };
 
-interface State {
+export interface State {
```
export not used?

szymonlesisz commented on packages/suite/src/components/suite/modals/AddAccount/components/SelectNetwork.tsx:
```diff
@@ -1,7 +1,7 @@
 import styled, { css } from 'styled-components';
 import { Icon, P, variables } from '@trezor/components';
-import { FADE_IN } from '@trezor/components/lib/config/animations';
+import { FADE_IN } from '@trezor/components/src/config/animations';
```
this belongs to previous commit

szymonlesisz commented on packages/suite/src/components/suite/Labeling/components/Metadata/index.tsx:
```diff
@@ -278,11 +278,10 @@ const MetadataLabeling = (props: Props) => {
                 setShowSuccess(true);
             }
-            timeout = setTimeout(() => {
+            timeout = window.setTimeout(() => {
```
is window reference required?

Nodonisko replied:
fixed

szymonlesisz commented on packages/suite/src/components/suite/BundleLoader/index.tsx:
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 
 import { Image } from '@suite-components';
-import { FADE_IN } from '@trezor/components/lib/config/animations';
+import { FADE_IN } from '@trezor/components/src/config/animations';
```
this belongs to previous commit

szymonlesisz commented on packages/suite/src/components/suite/Backdrop/index.tsx:
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
 import styled, { css } from 'styled-components';
-import { FADE_IN } from '@trezor/components/lib/config/animations';
+import { FADE_IN } from '@trezor/components/src/config/animations';
```
this belongs to previous commit

szymonlesisz commented on packages/blockchain-link/src/workers/base.ts:
```diff
@@ -11,11 +11,8 @@ import { WorkerState } from './state';
-// declare worker types locally
-// tsconfig is using "dom" in compilerOptions.
-declare const WorkerGlobalScope: ObjectConstructor | undefined;
+// self is not declared in TS Webworker lib typings
```
code history hygiene:
not sure if this blockchain-link change shouldnt be done in separated commit.
This fix was done by changing "web" > "webworker" not because of composite structure

Nodonisko replied:
done

szymonlesisz commented on packages/suite-build/package.json:
```diff
@@ -19,10 +19,11 @@
         "type-check:watch": "yarn type-check -- --watch"
     },
     "dependencies": {
+        "@trezor/suite": "1.0.0",
```
is it required?

Nodonisko replied:
it is, there is package.json from suite imported somewhere in this package

szymonlesisz commented on packages/suite-data/scripts/list-duplicates.ts:
```diff
@@ -1,4 +1,8 @@
+// this is causing module cyclic dependency!!! it will work since this dep is not specified
+// in package.json or refs, and also these files must be excluded from TS check
+// TODO: intl or scripts should have dedicated package on that could both suite-data and suite depends
+import messages from '@trezor/suite/src/support/messages';
```
totally agree!

szymonlesisz commented on packages/suite-desktop/tsconfig.json:
```diff
@@ -1,15 +1,16 @@
-    "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
-    "exclude": ["**/*.js", "src-electron/**/*.ts"]
+    "include": [
+        "next-env.d.ts",
```
next-env could be removed (doesn't exist)

szymonlesisz commented on packages/suite/package.json:
```diff
@@ -46,7 +46,7 @@
         "react-hook-form": "6.15.7",
         "react-intl": "^5.23.0",
-        "react-markdown": "^7.1.1",
+        "react-markdown": "^8.0.0",
```
is it required in this commit?

szymonlesisz commented on packages/suite/package.json:
```diff
@@ -26,7 +26,7 @@
         "@trezor/suite-data": "1.0.0",
         "@trezor/suite-storage": "1.0.0",
-        "@trezor/utils": "*",
+        "@trezor/utils": "1.0.0",
```
will be updated to ^1.1.0

szymonlesisz commented on packages/suite/src/components/suite/NavigationBar/components/DeviceSelector/index.tsx:
```diff
@@ -137,7 +137,7 @@ const DeviceSelector = (props: React.HTMLAttributes<HTMLDivElement>) => {
         // if the device status changes, show wallet state (dis/connected) as text for 2 seconds
         setShowTextStatus(true);
-        stateAnimationTimerRef.current = setTimeout(() => {
+        stateAnimationTimerRef.current = window.setTimeout(() => {
```
is window required?

szymonlesisz commented on packages/transport/package.json:
```diff
@@ -14,22 +14,23 @@
         "@types/jest": "^27.0.1",
-        "@types/w3c-web-usb": "^1.0.5"
+        "@types/w3c-web-usb": "^1.0.5",
+        "@trezor/utils": "1.0.0"
```
utils already declared in dependencies

szymonlesisz commented on packages/utils/package.json:
```diff
@@ -5,21 +5,23 @@
     "homepage": "https://github.com/trezor/trezor-suite/packages/utils",
     "description": "A collection of typescript utils that are intended to be used across trezor-suite monorepo.",
     "license": "SEE LICENSE IN LICENSE.md",
+    "private": true,
```
this package is not private, it is used as not direct dependency of trezor-connect, via blockchain-link for example

mroz22 replied:
definitely not private.

Nodonisko replied:
fixed

szymonlesisz commented on tsconfig.aliases.json:
```diff
@@ -0,0 +1,141 @@
+            "@desktop-electron/*": ["./packages/suite-desktop/src-electron/*"]
+        }
+    }
+}
```
this belongs to previous commit

szymonlesisz commented on yarn.lock:
```diff
@@ -7,6 +7,13 @@
   resolved "https://registry.yarnpkg.com/7zip-bin/-/7zip-bin-5.1.1.tgz#9274ec7460652f9c632c59addf24efb1684ef876"
   integrity sha512-sAP4LldeWNz0lNzmTird3uWfFDWWTeg6V/MsmyyLR9X1idwKBWIgt/ZvinqQldJm3LecKEs1emkbquO6PCiLVQ==
 
+"@ampproject/remapping@^2.0.0":
```
so many changes in yarn.lock in this commit? why?

Nodonisko replied:
should be fixed now

szymonlesisz commented on packages/blockchain-link/src/workers/electrum/client/json-rpc.ts:
```diff
@@ -1,5 +1,5 @@
 import { EventEmitter } from 'events';
-import { throwError } from '@trezor/utils';
+import { throwError } from '@trezor/utils/lib';
```
@trezor/utils is a public package. this fix and following should not be needed

Nodonisko replied:
agree, I already found better solution

Nodonisko replied:
fixed

szymonlesisz commented on packages/rollout/tsconfig.lib.json:
```diff
@@ -1,12 +1,11 @@
 {
     "extends": "./tsconfig.json",
     "compilerOptions": {
-        "module": "commonjs",
```
is this change required?

Nodonisko replied:
fixed

Nodonisko commented:
@szymonlesisz I refactored few things, mostly `trezor/utils` package. I also break commits into smaller ones, but in the end it's quite a lot of commits, maybe I can squash TS composite commits thats are now per package, but it makes more sense now.

matejkriz commented on packages/suite-web-landing/next.config.js:
```diff
@@ -1,10 +1,17 @@
+
+const { dependencies } = require('./package.json');
+
+// we definitely wan't to transpile all our local packages
```
Even though we don't use them all here?

matejkriz commented on packages/suite/src/utils/suite/__tests__/env.test.ts:
```diff
@@ -1,7 +1,14 @@
 import * as fixtures from '../__fixtures__/env';
 
-describe('isMacOs', () => {
+// TODO: this tests should be reworked due to mocking problem because of compilation.
```
We should probably create an issue for it at the time of merging those skips in develop.

matejkriz replied:
https://github.com/trezor/trezor-suite/issues/4993

Nodonisko commented on packages/suite-desktop/scripts/build.js:
```diff
@@ -44,12 +44,17 @@ console.log('[Electron Build] Starting...');
 console.log(`[Electron Build] Mode: ${isDev ? 'development' : 'production'}`);
 console.log(`[Electron Build] Using mocks: ${useMocks}`);
 
+// we want to bundle this
```
This is already solved in https://github.com/trezor/trezor-suite/pull/4797 by @mroz22

Nodonisko commented:
Rebased to fix conflicts

Nodonisko commented:
Rebased again + replaced `#` with `_` in `package.json` section names as we agreed on code improvements.

szymonlesisz commented on packages/utils/package.json:
```diff
@@ -12,16 +12,15 @@
-        "type-check": "tsc --project tsconfig.json",
-        "build:lib": "rimraf lib && tsc --p ./tsconfig.lib.json"
+        "type-check": "tsc --build tsconfig.json",
+        "build:lib": "rm -rf ./lib && tsc --build tsconfig.lib.json"
```
```suggestion
        "build:lib": "rimraf lib && tsc --build tsconfig.lib.json"
```

szymonlesisz commented on packages/blockchain-link/tsconfig.lib.json:
```diff
@@ -1,20 +1,15 @@
     },
-    "include": ["./src/index.ts", "./src/utils/ws.ts", "./src/utils/socks-proxy-agent.ts"],
-    "exclude": ["node_modules"]
+    "include": ["./src"]
```
or we can exclude `./src/ui` here since it's not needed in `lib` directory

Nodonisko replied:
in the end, I decided to keep it excluded only in `files` in package.json because if I want to exclude it here I will have to create quite big `exclude` that will include `lib, libDev, nodeModules, src/ui, build` etc. because using exclude here will override inherited one

szymonlesisz commented on packages/suite-build/tsconfig.json:
```diff
@@ -2,10 +2,8 @@
-    }
+        "outDir": "./libDev"
+    },
+    "include": ["**/*.ts", "**/*.tsx", "**/*.js"],
```
there shoudnt be any tsx files in this package (no react here)

Nodonisko replied:
simplified to just `"."`

szymonlesisz commented on packages/suite-desktop-api/tsconfig.lib.json:
```diff
@@ -1,17 +1,10 @@
     },
-    "include": ["./src/main.ts", "./src/renderer.ts"],
-    "exclude": ["node_modules"]
+    "include": ["./src"]
```
exclude unit `__tests__` and `tsconfig.lib.tsbuildinfo`

Nodonisko replied:
done, `.tsbuildinfo` is not produced at all for `lib` build now

szymonlesisz commented on packages/transport/package.json:
```diff
@@ -20,8 +20,8 @@
-        "type-check": "tsc --project tsconfig.json",
-        "build:lib": "rm -rf lib && tsc --p ./tsconfig.lib.json",
+        "type-check": "tsc --build tsconfig.json",
+        "build:lib": "rm -rf lib && tsc --build ./tsconfig.lib.json",
```
```suggestion
        "build:lib": "rimraf lib && tsc --build ./tsconfig.lib.json",
```

szymonlesisz commented on packages/transport/tsconfig.lib.json:
```diff
@@ -1,17 +1,10 @@
     },
-    "include": ["./src/index.ts"],
-    "exclude": ["node_modules", "**/tests/**", "**/fixtures/**", "**/*.test.ts", "**/e2e/**"]
+    "include": ["./src"]
```
exclude `tsconfig.lib.tsbuildinfo`

szymonlesisz commented on packages/utils/tsconfig.lib.json:
```diff
@@ -1,17 +1,11 @@
     },
-    "include": ["./src/index.ts", "./src/global.d.ts"],
-    "exclude": ["node_modules", "**/__tests__/**", "**/__fixtures__/**", "**/*.test.ts"]
+    "include": ["./src"]
```
exclude `tsconfig.lib.tsbuildinfo`

szymonlesisz commented on packages/utxo-lib/tsconfig.lib.json:
```diff
@@ -1,17 +1,12 @@
     },
-    "include": ["./src/index.ts", "./src/global.d.ts"],
-    "exclude": ["node_modules", "**/__tests__/**", "**/__fixtures__/**", "**/*.test.ts"]
+    "include": ["./src"]
```
exclude `tsconfig.lib.tsbuildinfo`

Nodonisko commented:
@szymonlesisz 

1. Rebased
2. Created new default `tsconfig.lib.json` in root that is used to extend `tsconfig.lib.json` in packages, which allowed me to simplify configs even more
3. Added `!**/*.map` to `files` in all published packages `package.json`
4. Simplified most of `tsconfig.json`  to us just `includes: ["."]`
5. Use `rimraf` in all packages

szymonlesisz commented on packages/rollout/tsconfig.lib.json:
```diff
@@ -1,18 +1,13 @@
 {
-    "compileOnSave": false,
+    "extends": "./tsconfig.json",
```
missed extend + related changes
```suggestion
    "extends": "../../tsconfig.json",
```

Nodonisko replied:
used correct default config

szymonlesisz commented on packages/suite/package.json:
```diff
@@ -72,6 +72,7 @@
         "zxcvbn": "^4.4.2"
     },
     "devDependencies": {
+        "@babel/preset-typescript": "^7.16.7",
```
`@babel/preset-typescript` (v 7.13.0) is already installed in top level package.json, do we have to have it here explicitly?

Nodonisko replied:
true, removed

szymonlesisz commented on packages/suite/jest.config.js:
```diff
@@ -68,12 +76,13 @@ module.exports = {
-        '(d3-|internmap).*\\.js$': 'ts-jest',
-        '\\.(ts|tsx)$': 'ts-jest',
+        '(d3-|internmap).*\\.js$': ['babel-jest', babelConfig],
+        '\\.(ts|tsx)$': ['babel-jest', babelConfig],
```
lets create followup issue to replace ts-jest with babel-jest in all packages

Nodonisko replied:
https://github.com/trezor/trezor-suite/issues/4981

szymonlesisz commented on packages/suite-desktop/package.json:
```diff
@@ -177,6 +175,7 @@
     "dependencies": {
         "@trezor/suite-desktop-api": "1.0.0",
         "@trezor/utils": "*",
+        "@trezor/suite": "1.0.0",
```
alphabetical order @trezor/utils should be after @trezor/suite

Nodonisko replied:
fixed

szymonlesisz commented on packages/suite-build/package.json:
```diff
@@ -19,10 +19,11 @@
         "type-check:watch": "yarn type-check -- --watch"
     },
     "dependencies": {
+        "@trezor/suite": "1.0.0",
```
alphabetical order @sentry before @trezor

szymonlesisz commented on packages/connect-explorer/package.json:
```diff
@@ -34,7 +34,8 @@
         "styled-components": "^5.3.3",
-        "trezor-connect": "8.2.7-beta.4"
+        "trezor-connect": "8.2.7-beta.4",
+        "@trezor/components": "*"
```
alphabetical order, @trezor should be at the top

Nodonisko replied:
fixed

szymonlesisz commented on packages/suite-desktop-api/package.json:
```diff
@@ -16,14 +16,15 @@
         "lib/main.js",
-        "lib/renderer.js"
+        "lib/renderer.js",
+        "**/*.map"
```
```suggestion
        "!**/*.map"
```

Nodonisko replied:
good catch! fixed

szymonlesisz reviewed (approved):
looks perfect now, i've briefly checked it locally and works great in dev mode

Nodonisko commented:
Fixed problem with including lot of unccessary node_modules and created followup task for some better and permanent solution => https://github.com/trezor/trezor-suite/issues/4989
